### PR TITLE
Remove undocumented Windows APIs (NtCreateKeyedEvent, NtReleaseKeyedEvent, NtWaitForKeyedEvent) in OpenJDK.

### PR DESCRIPTION
**Pull request by dhanalla**
Backport of https://github.com/openjdk/jdk/pull/20682 to JDK21

### CODE REVIEW

**Review by karianna — APPROVED** [review]
Looks clean, I assume we have manual tests covering this or is it possible to write some jtreg tests?

**Comment by dhanalla** [comment]
> Looks clean, I assume we have manual tests covering this or is it possible to write some jtreg tests?

Thanks @karianna, we have following test coverage 
1. JTreg - test/jdk/java/nio/channels/Selector
2. micro bench- test/micro/org/openjdk/bench/java/nio/channels
3. wepoll tests (https://github.com/piscisaureus/wepoll/tree/master/test) - These are wepoll tests to validate general scenarios such as poll, reference lock, and socket events (close, connect, open, fail), but they do not validate JVM scenarios.

File: ms-patches/remove-undocumentedAPI.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+title: Remove undocumented windows APIs
```
Comment (brianjstafford) [inline ms-patches/remove-undocumentedAPI.yaml:1]: Please capitalize Windows

**Review by brianjstafford — APPROVED** [review]
LGTM - just fix the capitalization of Windows

File: ms-patches/remove-undocumentedAPI.yaml (hunk 1 above)
Comment (dhanalla) [reply ms-patches/remove-undocumentedAPI.yaml]: Done

**Comment by dhanalla** [comment]
> LGTM - just fix the capitalization of Windows

Thank you @brianjstafford

**Pull request merged**